### PR TITLE
Fixed HTML & CSV export of non-BMP UTF-8

### LINKED ISSUE #1197: HTML/CSV export corrupts UTF-8 characters outside of Basic Multilingual Pane (BMP) ie code point >10000
OpenRefine 2.7 rc2

After reading UTF 8 file and executing export as UTF 8 file, garbled characters occurred.

displayed characters
image
有限会社なべ茶屋あさ𡌛

Exported garbled characters
image
有限会社なべ茶屋あさ����

other garbled export charactor sample
𣘺𣳾

### PR DESCRIPTION
**Pull request by tfmorris**
Fixes #1197

### CODE REVIEW

**Comment by thadguidry** [comment]
@tfmorris 
1. So was Excel "not broken" and only HTML and CSV exporter? (need to verify with you while I'm testing this on my Windows 10)
2. Hmm, also, to properly test...Do we need to have the option `-Dfile.encoding="UTF-8"` or is that entirely optional now when using OpenRefine on Windows?

File: main/src/com/google/refine/commands/project/ExportRowsCommand.java [hunk 1]
```diff
@@ -100,6 +100,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
                 contentType = exporter.getContentType();
             }
             response.setHeader("Content-Type", contentType);
```
Comment (thadguidry) [inline main/src/com/google/refine/commands/project/ExportRowsCommand.java:102]: Did I miss something?  but L102 still has Content-Type, even though you mentioned and changed to Content-Disposition ... was L102 missed as a deleted line?

**Comment by thadguidry** [comment]
Line 19 in his original file (UTF-8-BOM):
`10000～10FFF	𐀀 𐀁 𐅃 𐅄 𐌰 𐌱 𐌲 𐌳 𐌴 𐎀 𐎁 𐎈 𐎌 𐎑 𐎒 𐎗 𐎘 𐏏 𐿿 𐤀 𐤇 𐤈 𐤎	線文字B音節文字,古代ギリシア数字,ゴート文字,フェニキア文字`

When I import his original file and then export as CSV:
`"10000ï½ž10FFF	ð�€€ ð�€� ð�…ƒ ð�…„ ð�Œ° ð�Œ± ð�Œ² ð�Œ³ ð�Œ´ ð�Ž€ ð�Ž� ð�Žˆ ð�ŽŒ ð�Ž‘ ð�Ž’ ð�Ž— ð�Ž˜ ð��� ð�¿¿ ð�¤€ ð�¤‡ ð�¤ˆ ð�¤Ž	ç·šæ–‡å­—BéŸ³ç¯€æ–‡å­—,å�¤ä»£ã‚®ãƒªã‚·ã‚¢æ•°å­—,ã‚´ãƒ¼ãƒˆæ–‡å­—,ãƒ•ã‚§ãƒ‹ã‚­ã‚¢æ–‡å­—"`

**Comment by thadguidry** [comment]
Last line inside OpenRefine in FF browser (cell value copy):
`E0000～E0FFF	󠀀 󠀁 󠄀 󠄁 󠄂 󠄎 󠄏 󠌀 󠔀 󠜀 󠤀 󠨀 󠬀 󠰀 󠴀 󠸀 󠼀	タグ,字形選択子補助`

And exported as CSV:
`"E0000～E0FFF	󠀀 󠀁 󠄀 󠄁 󠄂 󠄎 󠄏 󠌀 󠔀 󠜀 󠤀 󠨀 󠬀 󠰀 󠴀 󠸀 󠼀	タグ,字形選択子補助"`

So this is good, the comma was retained, and CSV exporter wrapped with double-quotes correctly.

HTML export :
(last line)
`<tr><td>E0000～E0FFF	󠀀 󠀁 󠄀 󠄁 󠄂 󠄎 󠄏 󠌀 󠔀 󠜀 󠤀 󠨀 󠬀 󠰀 󠴀 󠸀 󠼀	タグ,字形選択子補助</td></tr>`

(and line 19)
`<tr><td>12000～12FFF	𒀀 𒀁 𒀂 𒀃 𒀄 𒀅 𒀆 𒀇 𒀈 𒀉 𒀊 𒀋 𒀌 𒀍 𒀎 𒀏 …	楔形文字,楔形文字の数字及び句読点</td></tr>`

@tfmorris I think we're golden now for this fix.  Great Job. 🥇

File: main/src/com/google/refine/commands/project/ExportRowsCommand.java (hunk 1 above)
Comment (tfmorris) [reply main/src/com/google/refine/commands/project/ExportRowsCommand.java:102]: The removed content type of `application/x-unknown` was in exporters.js. We still want to set the real content type e.g. `text/html`

**Comment by tfmorris** [comment]
@thadguidry Thanks for testing! Sorry for the false start, but both this and #2711 were updating the various exporters, so it really made more sense to combine the review. I'm closing this and replacing it and #2711 with #2713.

**Pull request closed without merging**